### ISSUE TITLE
Accumulate Gradients

### ISSUE THREAD

**Issue by alexandres** [clab/dynet#1478]
If my mini-batch is too big to fit in memory, how can I accumulate gradients over sub mini-batches? I tried:

```
foreach mb:
    foreach submb:
        dy.renew_cg()
        calculate loss
        loss.backward()
    trainer.step()
```

but I'm not getting the same results as when not using the sub mini-batches.

Just want to confirm that this is indeed correct so that I can look elsewhere in the model for the performance drop.

Thanks!

**Comment by alexandres** [comment]
Found the error in the model and now things are working correctly! :) 

The above is indeed the correct way to accumulate gradients and perform SGD on large mini-batches with limited GPU ram.
